vignettes/development_process.Rmd

```diff
@@ -0,0 +1,39 @@
+---
+title: "Development Process"
+output: 
+  rmarkdown::html_vignette
+    
+vignette: >
+  %\VignetteIndexEntry{Development Process}
+  %\VignetteEngine{knitr::rmarkdown}
+---
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+# Step-by-Step Guide
+
+Once you’ve familiarised yourself with the admiral [contribution model](https://roche-gsk.github.io/admiral/articles/contribution_model.html) and you’re ready to make your first code contribution, this development process will help tie all the other detailed vignettes together to give you the simplest experience of helping to grow and enhance our codebase.
+
+1. To start, you will have either created an issue or commented on an existing issue to notify that you’d like to contribute code. Then one of the admiral core team will assign you to the issue. 
+2. Create a new feature branch within the development branch (devel) following the naming convention and pull the latest changes - as detailed on the [github usage](https://roche-gsk.github.io/admiral/articles/git_usage.html#working-with-feature-branches-1) guide 
+3. Familiarise yourself with the admiral [programming strategy](https://roche-gsk.github.io/admiral/articles/programming_strategy.html), and then make the required code updates
+4. Before making a pull request , check the [Pull Request Review Guidance](https://roche-gsk.github.io/admiral/articles/pr_review_guidance.html) & the following checklist of common things developers miss:
+  a. Is all your code formatted according to the [tidyverse](https://style.tidyverse.org/) style guide? 
+  b. Did you create/add appropriate [unit tests](https://roche-gsk.github.io/admiral/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)?
+  c. If you removed/replaced any function did you fully follow the [deprecation guidance](https://roche-gsk.github.io/admiral/articles/programming_strategy.html#deprecation-1)?
+  d. Did you update the [documentation](https://roche-gsk.github.io/admiral/articles/programming_strategy.html#function-header-documentation-)? If so, remember to run `devtools::document()` and include the updated DESCRIPTION, NAMESPACE, and files in man/  
```

Comment:
I'm working on automating this step [here](https://github.com/Roche-GSK/admiral/pull/778). That would then be one less thing to aorry about.

Reply:
@thomas-neitmann that would be great! i could leave this in for now and update as soon as this automation is available or if you prefer i could just delete "d." altogether in anticipation of this. let me know if you prefer i update that now